include/net/net_pkt.h

```diff
@@ -900,6 +900,12 @@ static inline uint8_t *net_pkt_ip_data(struct net_pkt *pkt)
 	return pkt->frags->data;
 }
 
+static inline bool net_pkt_is_empty(struct net_pkt *pkt)
+{
+	return (!pkt->buffer || !net_pkt_data(pkt) ||
+		!pkt->buffer->len) ? true : false;
```

Comment:
I don't think this `? true : false` is needed here.

Reply:
Indeed, good point.

Reply:
Maybe use old good arithmetic comparison, `pkt->buffer->len == 0`? Friendly to humans, and even required by MISRA C.

Reply:
Well, sure, that can be done too.